src/diffusers/pipelines/flux/pipeline_flux_controlnet_inpainting.py

```diff
@@ -1095,11 +1095,13 @@ def __call__(
                 timestep = t.expand(latents.shape[0]).to(latents.dtype)
 
                 # predict the noise residual
-                if self.controlnet.config.guidance_embeds:
-                    guidance = torch.full([1], guidance_scale, device=device, dtype=torch.float32)
-                    guidance = guidance.expand(latents.shape[0])
-                else:
-                    guidance = None
+                if isinstance(self.controlnet, FluxMultiControlNetModel):
+                    use_guidance = self.controlnet.nets[0].config.guidance_embeds
+                    guidance = torch.tensor([guidance_scale], device=device) if use_guidance else None
+                else :
+                    use_guidance = self.controlnet.config.guidance_embeds
+                    guidance = torch.full([1], guidance_scale, device=device, dtype=torch.float32) if use_guidance else None
+                    guidance = guidance.expand(latents.shape[0]) if use_guidance else None
```

Comment:
```suggestion
                if isinstance(self.controlnet, FluxMultiControlNetModel):
                    use_guidance = self.controlnet.nets[0].config.guidance_embeds
                else:
                    use_guidance = self.controlnet.config.guidance_embeds
                if use_guidance:
                    guidance = torch.full([1], guidance_scale, device=device, dtype=torch.float32)
                    guidance = guidance.expand(latents.shape[0])
                else:
                    guidance = None
```

This is probably clearer.